aim/sdk/index_manager.py

```diff
@@ -0,0 +1,180 @@
+import time
+import click
+import tqdm
+import logging
+import os
+
+from threading import Thread
+from multiprocessing.pool import ThreadPool
+from psutil import cpu_count
+from filelock import FileLock
+from pathlib import Path
+from functools import partial
+
+from typing import Optional, List, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from aim.sdk.repo import Repo
+
+from aim.sdk.run import Run
+from aim.storage.rockscontainer import RocksContainer
+
+logger = logging.getLogger(__name__)
+
+
+class RepoIndexManager:
+    index_manager_pool = {}
```

Comment:
why do we need a pool though? isn't the manager going to be alive through the whole `aim up` up time, and will we have more than one manager inside the same process?

Reply:
@mihran113 the point is to have a single instance of `RepoIndexManager` per repo. Regardless how many times it's needed (for example API call to check indexing status should not create a new instance).
But your point is valid, since at the moment we do not have a use-case with more than one `Repo`.
Maybe worth considering a `Singleton`? 🤔

Reply:
The other option is to add `IndexManager` instance to the members of `Project` class, and reuse it whenever needed

Reply:
@mihran113 that would create a dependency between cli `reindex` command and `web` module.

Reply:
let's keep it now as it is then